src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/CannotRemoveBaseTypeOrInterface.cs

```diff
@@ -85,6 +100,23 @@ private void ValidateInterfaceNotRemoved(ITypeSymbol left, ITypeSymbol right, st
                     return;
                 }
             }
+
+            foreach (ITypeSymbol rightInterface in rightInterfaces)
+            {
+                if (rightInterface.TypeKind == TypeKind.Error && Settings.WithReferences)
+                {
+                    AddAssemblyLoadError(differences, rightInterface);
+                }
+            }
+        }
+
+        private void AddAssemblyLoadError(IList<CompatDifference> differences, ITypeSymbol type)
+        {
+            differences.Add(new CompatDifference(
```

Comment:
Will this ensure only a single diagnostic per assembly?

Reply:
it won't. Good catch, I'll fix this and add a test for it.